disco_aws_automation/disco_aws.py

```diff
@@ -308,21 +306,95 @@ def update_elb(self, hostclass, update_autoscaling=True, testing=False):
                 elb_subnets = self.get_subnets(elb_meta_network, hostclass)
                 subnet_ids = [subnet['SubnetId'] for subnet in elb_subnets]
 
-            elb_port = self.hostclass_option_default(hostclass, "elb_port", 80)
-            elb_protocol = self.hostclass_option_default(hostclass, "elb_protocol", None) or \
-                self._default_protocol_for_port(elb_port)
-            instance_port = int(self.hostclass_option_default(hostclass, "elb_instance_port", 80))
-            instance_protocol = self.hostclass_option_default(hostclass, "elb_instance_protocol", None) or \
-                self._default_protocol_for_port(instance_port)
+            def default_protocol_for_port(port):
+                """
+                Return the default protocl for the port
+                """
+                return {80: 'HTTP', 443: 'HTTPS'}.get(int(port), 'TCP')
+
+            def default_port_for_protocol(protocol):
+                """
+                Return the default port for the protocol
+                """
+                return {'HTTP': 80, 'HTTPS': 443}[protocol]
+
+            def unzip(things):
+                """
+                The inverse of zip
+                """
+                return zip(*things)
```

Comment:
That was my reaction when I first saw this recipe, which is why I wrapped in a function :)